lib/spack/spack/build_systems/python.py

```diff
@@ -222,6 +223,17 @@ def install(self, spec, prefix):
 
         self.setup_py('install', *args)
 
+        rpaths = []
+        rpaths.append(spec.prefix)
+        for d in spec.dependencies(deptype=('link', 'run')):
+            if re.match('^py-', d.name):
```

Comment:
is this sufficient for all python dependencies? There are packages which do `extends('python')` but are not named `py-`

Reply:
No, it's not sufficient.  To a first approx. we'd need to wire up `.spack-rpath` files for the same set of things that set PYTHONPATH in their modules.  Probably the right way to do that is to grab it out of `setup_dependent_environment` (copying the mechanism that the module system uses).

Reply:
No, relying on the name is a bad idea.  You need to check for whether the package `extends('python')`.  I don't know how to do this check, but there must be a way.  And it is probably the right way.

Reply:
Yep, if folks think this approach is worth pursuing I'm happy to grind out the details.